File: 프로그래머스/0/181891. 순서 바꾸기/순서 바꾸기.java
import java.util.*;

class Solution {
    public int[] solution(int[] num_list, int n) {
//         int[] first = Arrays.copyOfRange(num_list, 0, n);   //2
//         int[] second = Arrays.copyOfRange(num_list, n, num_list.length);    //1, 6
        
//         int[] answer = new int[num_list.length];
        
//         System.arraycopy(first, 0, answer, second.length, first.length);    //x, x, 2
//         System.arraycopy(second, 0, answer, 0, second.length);
//         return answer;
        
        int[] answer = new int[num_list.length];
        int index = 0;
        for(int i=n; i<num_list.length; i++) {  //1, 6
            answer[index++] = num_list[i];  //[0]=1, [1]=6
        }
        
        for(int i=0; i<n; i++) {
            answer[index++] = num_list[i];
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181891. 순서 바꾸기/순서 바꾸기.js
function solution(num_list, n) {
    // let answer = num_list.slice(n);
    // for(let i=0; i<n; i++) {
    //     answer.push(num_list[i]);
    // }
    // return answer;
    
    // let start = num_list.slice(n);
    // let end = num_list.slice(0, n);
    // return start.concat(end);
    
    let start = num_list.slice(n);
    let end = num_list.slice(0, n);
    return [...start, ...end];
    
    // num_list.push(...num_list.splice(0, n));
    // return num_list;
}